sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1047,10 +1104,10 @@ private void releaseChannel(final Channel channel, final Promise<Void> promise)
 
         checkState(channel.eventLoop().inEventLoop());
 
-        final ChannelPool pool = channel.attr(POOL_KEY).getAndSet(null);
-        final boolean acquired = this.acquiredChannels.get(channel) != null;
+        final RntbdClientChannelPool pool = channel.attr(POOL_KEY).getAndSet(null);
+        final Channel acquiredChannel = this.acquiredChannels.get(channel);
 
-        if (acquired && pool == this) {
+        if (acquiredChannel == channel && pool == this) {
```

Comment:
nit: personally I like the acquired boolean pattern better.
This pattern assumes that non-null other values are possibles.